cibyl/config.py

```diff
@@ -43,27 +44,72 @@ def _ask_user_for_overwrite():
     and the app.
     """
-
-    def load(self, file):
-        """Loads the contents of a file into this object. Any contents this
-        may beforehand have are lost and replaced by the data from the file.
-        In case of error, the contents are left untouched.
-
-        :raises YAMLError: If the file could not be parsed.
+    def load(self, path=None, data=None, file=None):
+        """Loads the content of a configuration file/object and creates
+        a reference to environments."""
+        if data:
+            self.data = data
+        elif path:
+            self.data = ConfigFactory.from_path(path)
+        elif file:
+            self.data = ConfigFactory.from_file(file)
+        else:
+            self.data = ConfigFactory.from_search()
+
+
+class AppConfig(Config):
+    """Representation of a Cybil's configuration file"""
+
+    def __init__(self):
+        """AppConfig Constructor"""
+        super(Config, self).__init__()
+        self.environments = {}
+        self.plugins = {}
+
+    def load(self, path=None, data=None):
+        """Loads the content of a configuration file/object and creates
+        a reference to environments."""
+        super().load(path, data)
+        self.environments = self.data.get('environments', {})
+        self.plugins = self.data.get('plugins', [])
+        self.verify_environments()
+        self.verify_systems()
+
+    def verify_environments(self) -> None:
+        """Checks whether the environments section in the configuration
+        is valid.
+        :param env_section: The environments section
+        :type path: dict, str or None
+        :raise MissingSystems: If a specific environment section is empty
+        :raise MissingEnvironments: If no environments specified
         """
-        self.data = yaml.parse(file)
-        if self.data is None:
-            # if the configuration file is empty, yaml.parse will return None,
-            # we assign an empty dictionary to always return the same type and
-            # raise an exception
-            self.data = {}
-            raise EmptyConfiguration(file)
+        if isinstance(self.environments, str):
+            raise conf_exc.MissingSystems(self.environments)
+        if not self.environments:
+            raise conf_exc.MissingEnvironments()
+
+    def verify_systems(self) -> None:
+        for env_name, systems_dict in self.environments.items():
+            if isinstance(systems_dict, str):
+                raise conf_exc.MissingSystemType(systems_dict, SystemType)
+            try:
+                systems_dict.get('foo', None)
+            except AttributeError:
+                raise conf_exc.MissingSystems(env_name)
+            for system_name, single_system in systems_dict.items():
+                self.verify_sources(system_name, single_system)
+
+    def verify_sources(self, system_name, system_data) -> None:
+        try:
+            system_data['sources']
```

Comment:
Result will be the same, but using an 'if' statement will avoid having to rely on an error case: `if 'sources' not in system_data: raise ...`